frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -421,6 +424,90 @@ final class BloopBspServices(
     Task.eval { originToCompileStores.remove(params.originId); () }.executeAsync
   }
 
+  def linkProjects(
+      userProjects: Seq[ProjectMapping],
+      state: State,
+      compileArgs: List[String],
+      originId: Option[String],
+      logger: BspServerLogger
+  ): BspResult[bsp.CompileResult] = {
+    import bloop.engine.tasks.LinkTask.{linkJS, linkNative}
+
+    val isRelease = compileArgs.exists(_ == "--release")
+    val isDebug = compileArgs.exists(_ == "--debug")
+
+    val overrideLinkerMode = if (isRelease && isDebug) {
+      None
```

Comment:
MAybe we should log an error in this case?

Reply:
I logged it as a warning as we still run the linking, we just ignore the dynamic --debug or --release arguments.